frontend/src/pages/modelServing/screens/projects/useIsNIMAvailable.ts

```diff
@@ -1,21 +1,31 @@
-import { useEffect, useState } from 'react';
+import * as React from 'react';
 import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
-import { isNIMServingRuntimeTemplateAvailable } from '~/pages/modelServing/screens/projects/nimUtils';
-import { useDashboardNamespace } from '~/redux/selectors';
+import { getIntegrationAppEnablementStatus } from '~/services/integrationAppService';
+import { fetchComponents } from '~/services/componentsServices';
+import useFetchState from '~/utilities/useFetchState';
 
-export const useIsNIMAvailable = (): boolean => {
-  const { dashboardNamespace } = useDashboardNamespace();
-  const [isNIMAvailable, setIsNIMAvailable] = useState<boolean>(false);
+export const useIsNIMAvailable = (): [boolean, boolean, Error | undefined] => {
   const isNIMModelServingAvailable = useIsAreaAvailable(SupportedArea.NIM_MODEL).status;
 
-  useEffect(() => {
-    const checkNIMServingRuntime = async () => {
-      const isNIMRuntimeAvailable = await isNIMServingRuntimeTemplateAvailable(dashboardNamespace);
-      setIsNIMAvailable(isNIMModelServingAvailable && isNIMRuntimeAvailable);
-    };
+  const fetchNIMAvailability = React.useCallback(async () => {
+    const components = await fetchComponents(false);
```

Comment:
Could you make the `fetchNIMAvailability` return early with `false` if `isNIMModelServingAvailable` is `false`? This way it will avoid the following API calls when we already know nim is not available

Reply:
Updated